src/SwedbankPay.Sdk/PaymentOrders/PaymentOrder.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using SwedbankPay.Sdk.Payments;
+using SwedbankPay.Sdk.Transactions;
+
+namespace SwedbankPay.Sdk.PaymentOrders
+{
+    public class PaymentOrder
+    {
+        private PaymentOrder(PaymentOrderResponseContainer paymentOrderResponseContainer,
+                             SwedbankPayHttpClient client)
+        {
+            PaymentOrderResponse = paymentOrderResponseContainer.PaymentOrderResponse;
+            var operations = new Operations();
+
+            foreach (var httpOperation in paymentOrderResponseContainer.Operations)
+            {
+                operations.Add(httpOperation.Rel, httpOperation);
+
+                switch (httpOperation.Rel.Value)
+                {
+                    case PaymentOrderResourceOperations.CreatePaymentOrderCapture:
+                        operations.Capture =
+                            new ExecuteRequestWrapper<TransactionRequestContainer, CaptureTransactionResponseContainer>(
```

Comment:
Consider finding a more readable way to do this. Essentially the code here is just Func<TRequest, TResponse>. What we currently have is a custom class with two different versions, using inheritance, and duplicating code already found in SwedbankPayHttpClient.

ExecuteWrapperBase has code to update payload and headers on a request. This code already exists when creating a request in SwedbankPayHttpClient and has been duplicated. I suggest we somehow enforce these headers when constructing the request, e.g. with the help of a static factory method. This would also mean the code in SwedbankPayHttpClient that is used to create requests needs to be deleted. Additionally, attaching the payload (and settings its content type) should also be extended onto the request class itself somehow so that it's not just loosely laying around like it is now. In conclusion, we don't really need ExecuteWrapperBase.

The two classes ExecuteRequestWrapper and ExecuteWrapper can be replaced by Func<TRequest, TResponse> and Func<TResponse> respectively. After all, what we want is a wrapper. If we're not doing anything significant in the wrapper we might as well use a simpler kind of wrapper. All we really want to do in the wrapper is to call the SwedbankPayHttpClient's "Send this request(request here)", a single line.

Reply:
Another thing to mention is that HttpOperation's request Accept header gets overwritten when it's put into an ExecuteWrapper, making the contentType argument of the constructor potentially useless.

Reply:
@muzukashiine we have talked to @asbjornu regarding this. Please discuss this with him a bit first.